iron-focus-wrap-behavior.html

```diff
@@ -0,0 +1,236 @@
+<!--
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-a11y-keys-behavior/iron-a11y-keys-behavior.html">
+
+<script>
+(function() {
+'use strict';
+
+/**
+`Polymer.IronFocusWrapBehavior` can wrap (and trap) the focus within an element
+as soon as it gets focused, by listening at `TAB` keyboard events.
+
+When `TAB` is pressed and the `lastFocusable` is already focused, `IronFocusWrapBehavior`
+will move the focus to `firstFocusable`.
+Likewise, when `Shift+TAB` is pressed and
+the `firstFocusable` is already focused, `IronFocusWrapBehavior` will move
+the focus to `lastFocusable`.
+
+    <button id="first">first</button>
+    <content></content>
+    <button id="last">last</button>
+
+    ...
+
+    ready: function() {
+      this.wrapFocus = true;
+      this.firstFocusable = this.$.first;
+      this.lastFocusable = this.$.last;
+    }
+
+In addition, `Polymer.IronFocusWrapBehavior` can be used to save and restore the
+focus of an active element.
+
+    // Before moving the focus, store the deep active element.
+    this.saveDeepActiveElement();
+    this.focus();
+
+    // later on...
+    this.blur();
+    this.restoreDeepActiveElement();
+
+@demo demo/index.html
+@polymerBehavior Polymer.IronFocusWrapBehavior
+*/
+
+  Polymer.IronFocusWrapBehaviorImpl = {
+
+    properties: {
+      /**
+       * The first focusable element. It should be a child of the element or the
+       * element itself. Defaults to `this`.
+       * @type {Element}
+       */
+      firstFocusable: {
+        type: Element,
+        value: function() {
+          return this;
+        }
+      },
+      /**
+       * The last focusable element. It should be a child of the element or the
+       * element itself. Defaults to `this`.
+       * @type {Element}
+       */
+      lastFocusable: {
+        type: Element,
+        value: function() {
+          return this;
+        }
+      },
+      /**
+       * If true, it will wrap the focus on `TAB` or `Shift+TAB` when the focus
+       * is already in the element.
+       */
+      wrapFocus: Boolean
```

Comment:
That would require an observer on the property, and we wouldn't benefit too much from removing the key binding (as the event listener will still be active)